### ISSUE TITLE
33.1.1.700 -- Android -- ADT fails to launch App after install

### ISSUE THREAD

**Issue by urthling** [airsdk/Adobe-Runtime-Support#1430]
Normally, when debugging on Android, the App is launched after it is installed on the device.  With latest `adt` although AIR and the App are installed successfully, the App is not launched automatically.

Workaround is simply to manually launch the App by clicking the icon (as we do on iOS)

**Comment by ajwfrost** [comment]
Hi @urthling, what IDE are you using currently? From the command-line tools, the install and launch are separate commands so this must be an IDE that's perhaps getting an unexpected result from the 'install' command.
Plus is it an APK or an AAB that you're generating?
thanks

**Comment by jimmymjing** [comment]
I'm having the same issue. I'm using IntelliJ and generating APK.

**Comment by urthling** [comment]
Still using Flashbuilder (hard to beat the multi-window layout!) and I'm generating an APK

**Comment by urthling** [comment]
I must say though, a development build now appears to be far quicker on Android, so kudos to you and the team for that!!

**Comment by ajwfrost** [comment]
So this issue is due to the change in packaging up APK files between the legacy mechanism (which would work) and the Gradle mechanism (which generates a slightly different main class name and hence the launch command isn't working...)

Will get this updated...

**Comment by marchbold** [comment]
@ajwfrost Not sure if you have broken the legacy build version now? I'm getting this behaviour with `BuildLegacyAPK` in 33.1.1.713. I can see an error in the logs attempting to start `air.com.distriqt.test/.AIRAppEntry` but works if I launch `air.com.distriqt.test/.AppEntry`: 

```
adb shell am start -n air.com.distriqt.test/.AppEntry
```

Am I correct in the following change for the main app activity:

legacy: `air.package.name/.AppEntry`
new: `air.package.name/.AIRAppEntry`


Not sure how you trigger it but you could use this method:

```
adb shell monkey -p air.package.name 1
```

which doesn't rely on the class name, just simulates the app icon click?

**Comment by ajwfrost** [comment]
Hi @marchbold  - the code is meant to try the new activity name (.AIRAppEntry) and if it fails, then to try again with the old (.AppEntry). I guess something went wrong with the fallback mechanism - it's trying to detect the adb output (and looks for "Error type 3"...) to discern between an 'activity not found' vs any other error.

Good suggestion to use that `monkey` thing but wouldn't this then constrain it to just the given package i.e. anything that relies on other packages from the Android system would then fail?

What do you see from the command-line if you do run:
```
adb shell am start -n air.com.distriqt.test/.AIRAppEntry
```

thanks

**Comment by marchbold** [comment]
```
❯ adb shell am start -n air.com.distriqt.test/.AIRAppEntry
Starting: Intent { cmp=air.com.distriqt.test/.AIRAppEntry }
Error type 3
Error: Activity class {air.com.distriqt.test/air.com.distriqt.test.AIRAppEntry} does not exist.
```

Note, those 2 lines with `Error` are output to stderr, while the other line is stdout.

**Comment by marchbold** [comment]
What do you mean by "constrain it to just the given package" ? This monkey command just simulates a normal launch from the app icon, it doesn't limit the packages i.e. it is the same as triggering the `android.intent.category.LAUNCHER` category for this package 

This is a more explicit version of the command:

```
adb shell monkey -p air.package.name -c android.intent.category.LAUNCHER 1
```

**Comment by marchbold** [comment]
Yeah you are right about that monkey command don't think it's really enough to use in this case.

**Comment by ajwfrost** [comment]
Thanks - I think the problem may be the mixture of stdout and stderr, will check that ...

In terms of the monkey command, from https://developer.android.com/studio/test/monkey:
<html>
<body>


-p <allowed-package-name> | If you specify one or more packages this way, the Monkey will only allow the system to visit activities within those packages. If your application requires access to activities in other packages (e.g. to select a contact) you'll need to specify those packages as well. If you don't specify any packages, the Monkey will allow the system to launch activities in all packages. To specify multiple packages, use the -p option multiple times — one -p option per package.
-- | --



</body>
</html>

**Comment by marchbold** [comment]
An alternative is to use `cmd package` to search for the launch activity name:

```
❯ adb shell cmd package resolve-activity -c android.intent.category.LAUNCHER air.com.distriqt.test
priority=0 preferredOrder=0 match=0x108000 specificIndex=-1 isDefault=false
ActivityInfo:
  name=air.com.distriqt.test.AIRAppEntry
  packageName=air.com.distriqt.test
  labelRes=0x7f110029 nonLocalizedLabel=null icon=0x0 banner=0x0
  enabled=true exported=true directBootAware=false
  taskAffinity=air.com.distriqt.test targetActivity=null persistableMode=PERSIST_ROOT_ONLY
  launchMode=2 flags=0x200 privateFlags=0x0 theme=0x7f120132
  screenOrientation=2 configChanges=0x40000da3 softInputMode=0x12
  lockTaskLaunchMode=LOCK_TASK_LAUNCH_MODE_DEFAULT
  resizeMode=RESIZE_MODE_RESIZEABLE_VIA_SDK_VERSION
  ApplicationInfo:
    name=androidx.multidex.MultiDexApplication
    packageName=air.com.distriqt.test
    labelRes=0x7f110029 nonLocalizedLabel=null icon=0x7f0e0000 banner=0x0
    className=androidx.multidex.MultiDexApplication
    processName=air.com.distriqt.test
    taskAffinity=air.com.distriqt.test
    uid=10791 flags=0x3088be44 privateFlags=0x8c001000 theme=0x0
    requiresSmallestWidthDp=0 compatibleWidthLimitDp=0 largestWidthLimitDp=0
    sourceDir=/data/app/~~WLi9fPXw5O8MUFaRlACLOA==/air.com.distriqt.test-ufbpfHFVcxpCieiqXitSeg==/base.apk
    resourceDirs=[/product/overlay/IconPackRoundedAndroid/IconPackRoundedAndroidOverlay.apk, /product/overlay/AccentColorBlack/AccentColorBlackOverlay.apk, /product/overlay/FontRubikRubik/FontRubikRubikOverlay.apk, /product/overlay/IconShapeSquircle/IconShapeSquircleOverlay.apk, /product/overlay/NavigationBarModeGestural/NavigationBarModeGesturalOverlay.apk]
    seinfo=default:targetSdkVersion=30
    seinfoUser=:complete
    dataDir=/data/user/0/air.com.distriqt.test
    deviceProtectedDataDir=/data/user_de/0/air.com.distriqt.test
    credentialProtectedDataDir=/data/user/0/air.com.distriqt.test
    enabled=true minSdkVersion=19 targetSdkVersion=30 versionCode=17 targetSandboxVersion=1
    supportsRtl=false
    fullBackupContent=true
    crossProfile=false
    HiddenApiEnforcementPolicy=2
    usesNonSdkApi=false
    allowsPlaybackCapture=true
```
